### PR TITLE
[BUG][Go] Remove "null" body value when body is empty #13927

### PR DESCRIPTION
**Pull request by rcebrian**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

## Summary
This PR fix the issue [13927](https://github.com/OpenAPITools/openapi-generator/issues/13927).
The changes affects directly to go-server generator (chi and mux). The ```routers.go``` mustache template has been modified including a new clause that checks if there is body, encode an include.

### Technical committees
* @antihax
* @grokify 
* @kemokemo 
* @jirikuncar 
* @ph4r5h4d

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/go-server/routers.mustache [hunk 1]
```diff
@@ -108,7 +108,11 @@ func EncodeJSONResponse(i interface{}, status *int,{{#addResponseHeaders}} heade
 		w.WriteHeader(http.StatusOK)
 	}
 
-	return json.NewEncoder(w).Encode(i)
+    if (i) != nil {
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go-server/routers.mustache:111]: what about just the following without parenthesis ?
```
if i != null {
```

Comment (rcebrian) [reply modules/openapi-generator/src/main/resources/go-server/routers.mustache]: Changed!

**Comment by wing328** [comment]
thanks for the PR. if no further feedback or question from anyone, I'll merge it tomorrow.

File: modules/openapi-generator/src/main/resources/go-server/routers.mustache [hunk 2]
```diff
@@ -108,7 +108,11 @@ func EncodeJSONResponse(i interface{}, status *int,{{#addResponseHeaders}} heade
 		w.WriteHeader(http.StatusOK)
 	}
 
-	return json.NewEncoder(w).Encode(i)
+    if i != nil {
+        return json.NewEncoder(w).Encode(i)
+    }
+
+    return nil
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go-server/routers.mustache:115]: FYI. Filed and merged https://github.com/OpenAPITools/openapi-generator/pull/14740 to replace 4-space with tabs

**Pull request merged**